homeassistant/components/caldav/todo.py

```diff
@@ -99,9 +99,11 @@ def _to_ics_fields(item: TodoItem) -> dict[str, Any]:
         item_data["status"] = TODO_STATUS_MAP_INV.get(status, "NEEDS-ACTION")
     if due := item.due:
         if isinstance(due, datetime):
-            item_data["due"] = dt_util.as_utc(due).strftime("%Y%m%dT%H%M%SZ")
+            item_data["due"] = dt_util.as_utc(due)
+        elif isinstance(due, date):
+            item_data["due"] = due
         else:
-            item_data["due"] = due.strftime("%Y%m%d")
+            item_data["due"] = dt_util.as_utc(dt_util.parse_datetime(due))
```

Comment:
This case shouldn't be possible given `due` is `datetime | date`

Reply:
Lets go with `item_data["due"] = due` to handle all cases not do any UTC conversions and preserve the local timezone.

Reply:
Thank you for the review! :)

I wasn't sure why the due date was converted to UTC, so I kept it. But I guess mainly due to the manual stringifying?
I hadn't checked the annotations of `due` I guess. :facepalm: